Doc comment bug with exceptions

The generated C# code for the Slice definition below fails to C# compile with this error:

```
 All projects are up-to-date for restore.
C:\Users\BernardNormier\source\repos\icerpc-csharp\src\IceRpc\Slice\generated\DispatchException.cs(29,31): error CS1573: Parameter 'errorCode' has no matching param tag in the XML comment for 'DispatchException.DispatchException(DispatchErrorCode, RetryPolicy?)' (but other parameters do) [C:\Users\BernardNormier\source\repos\icerpc-csharp\src\IceRpc\IceRpc.csproj]
C:\Users\BernardNormier\source\repos\icerpc-csharp\src\IceRpc\Slice\generated\DispatchException.cs(42,31): error CS1573: Parameter 'errorCode' has no matching param tag in the XML comment for 'DispatchException.DispatchException(string?, DispatchErrorCode, Exception?, RetryPolicy?)' (but other parameters do) [C:\Users\BernardNormier\source\repos\icerpc-csharp\src\IceRpc\IceRpc.csproj]
```

Here is the corresponding ctor:
```
        /// <summary>Constructs a new instance of <see cref="DispatchException"/>.</summary>
        /// <param name="retryPolicy">The retry policy for the exception.</param>
        public DispatchException(
            DispatchErrorCode errorCode,
            IceRpc.RetryPolicy? retryPolicy = null)
            : base(retryPolicy)
        {
            this.ErrorCode = errorCode;
        }
```

For some unknown reason, there is no param doc for the errorCode parameter.

```
// Copyright (c) ZeroC, Inc. All rights reserved.

[cs:namespace("IceRpc.Slice")]
module Slice;

/// A dispatch exception is a built-in exception thrown when a server fails to dispatch a request. Since it's built-in,
/// any client can decode this exception.
exception DispatchException
{
    /// The error code that describes the failure.
    errorCode: DispatchErrorCode,
}

/// The possible error codes carried by @DispatchException.
unchecked enum DispatchErrorCode
{
    /// The dispatch could not find the target service.
    ServiceNotFound,

    /// The dispatch found the target service but this target service does not have the requested operation.
    OperationNotFound,

    /// The dispatch failed with an unhandled exception.
    UnhandledException,

    /// The dispatch failed because the request header or the request payload could not be decoded.
    InvalidData,

    /// The dispatch was canceled.
    Canceled,
}

```

--- a/tools/slicec-cs/src/builders.rs
+++ b/tools/slicec-cs/src/builders.rs
@@ -222,13 +222,6 @@ impl FunctionBuilder {
         self
     }
 
-    pub fn add_parameters(&mut self, parameters: &[String]) -> &mut Self {
-        for p in parameters {
-            self.parameters.push(p.clone());
-        }
-        self
-    }
-
     pub fn add_base_parameter(&mut self, argument: &str) -> &mut Self {
         self.base_arguments.push(argument.to_owned());
         self

--- a/tools/slicec-cs/src/exception_visitor.rs
+++ b/tools/slicec-cs/src/exception_visitor.rs
@@ -12,7 +12,7 @@ use crate::generated_code::GeneratedCode;
 use crate::member_util::*;
 use crate::slicec_ext::*;
 use slice::code_gen_util::TypeContext;
-use slice::grammar::Exception;
+use slice::grammar::{Exception, Member};
 use slice::visitor::Visitor;
 
 pub struct ExceptionVisitor<'a> {
@@ -91,7 +91,9 @@ impl<'a> Visitor for ExceptionVisitor<'_> {
                 .set_body({
                     let mut code = CodeBlock::new();
                     if !has_base && !members.is_empty() && !exception_def.uses_classes() {
-                        writeln!(code, "\
+                        writeln!(
+                            code,
+                            "\
 if (decoder.Encoding == IceRpc.Encoding.Slice11)
 {{
     {initialize_non_nullable_fields}
@@ -101,11 +103,18 @@ else
     {decode_data_members}
 }}
                         ",
-                        initialize_non_nullable_fields =
-                            initialize_non_nullable_fields(&members, FieldType::Exception).indent(),
-                        decode_data_members = decode_data_members(&members, namespace,  FieldType::Exception).indent())
+                            initialize_non_nullable_fields =
+                                initialize_non_nullable_fields(&members, FieldType::Exception)
+                                    .indent(),
+                            decode_data_members =
+                                decode_data_members(&members, namespace, FieldType::Exception)
+                                    .indent()
+                        )
                     } else {
-                        code.writeln(&initialize_non_nullable_fields(&members, FieldType::Exception))
+                        code.writeln(&initialize_non_nullable_fields(
+                            &members,
+                            FieldType::Exception,
+                        ))
                     }
                     code
                 })
@@ -220,17 +229,6 @@ fn one_shot_constructor(
     let inner_exception_parameter_name = escape_parameter_name(&all_data_members, "innerException");
     let retry_policy_parameter_name = escape_parameter_name(&all_data_members, "retryPolicy");
 
-    let all_parameters = all_data_members
-        .iter()
-        .map(|m| {
-            let member_type = m
-                .data_type
-                .to_type_string(namespace, TypeContext::DataMember, false);
-            let member_name = m.parameter_name();
-            format!("{} {}", member_type, member_name)
-        })
-        .collect::<Vec<_>>();
-
     let base_parameters = if let Some(base) = exception_def.base_exception() {
         base.all_members()
             .iter()
@@ -261,7 +259,16 @@ fn one_shot_constructor(
         ctor_builder.add_base_parameter(&message_parameter_name);
     }
 
-    ctor_builder.add_parameters(&all_parameters);
+    for member in &all_data_members {
+        ctor_builder.add_parameter(
+            &member
+                .data_type()
+                .to_type_string(&namespace, TypeContext::DataMember, false),
+            member.parameter_name().as_str(),
+            None,
+            Some(&doc_comment_message(*member)),
+        );
+    }
     ctor_builder.add_base_parameters(&base_parameters);
 
     if add_message_and_exception_parameters {

